pkg/agent/manager/storecache/cache.go

```diff
@@ -250,6 +250,10 @@ func (c *Cache) GetStaleEntries() []*cache.StaleEntry {
 	return staleEntries
 }
 
+func (c *Cache) CountX509SVIDs() int {
+	return len(c.records)
```

Comment:
We should lock the mutex for reading to get the length of the records.
```suggestion
	c.mtx.RLock()
	defer c.mtx.RUnlock()

	return len(c.records)
```